tests/src/Unit/CheckFieldCardinalityTest.php

```diff
@@ -0,0 +1,143 @@
+  }
+
+  /**
+   * @covers ::checkFieldCardinality
+   *
+   * @expectedException \Drupal\Core\Field\FieldException
+   */
+  public function testFieldCardinalityNotAudienceField() {
+    $field_name = 'test_field_no_definition';
+    $entity_prophecy = $this->prophesize(ContentEntityInterface::class);
+
+    $field_definition_prophecy = $this->prophesize(FieldDefinitionInterface::class);
+    $field_definition_prophecy->getType()
+      ->willReturn('invalid_field_type')
+      ->shouldBeCalled();
+    $field_definition_prophecy->getFieldStorageDefinition()
+      ->shouldNotBeCalled();
+
+    $entity_prophecy->getFieldDefinition($field_name)
+      ->willReturn($field_definition_prophecy->reveal());
+
+    // The bundle() and getEntityTypeId() methods will be called for the
+    // exception string.
+    $entity_prophecy->bundle()
+      ->shouldBeCalled();
+    $entity_prophecy->getEntityTypeId()
+      ->shouldBeCalled();
+
+    OgGroupAudienceHelper::checkFieldCardinality($entity_prophecy->reveal(), $field_name);
+  }
+
+  /**
+   * @covers ::checkFieldCardinality
+   *
+   * @dataProvider providerTestFieldCardinality
+   */
+  public function testFieldCardinality($field_count, $cardinality, $expected) {
+    $field_name = 'test_field';
+
+    $field_storage_definition_prophecy = $this->prophesize(FieldStorageDefinitionInterface::class);
+    $field_storage_definition_prophecy->getCardinality()
+      ->willReturn($cardinality)
+      ->shouldBeCalled();
+
+    $field_definition_prophecy = $this->prophesize(FieldDefinitionInterface::class);
+    $field_definition_prophecy->getFieldStorageDefinition()
+      ->willReturn($field_storage_definition_prophecy->reveal())
+      ->shouldBeCalled();
+    $field_definition_prophecy->getType()
+      ->willReturn('og_membership_reference')
+      ->shouldBeCalled();
+
+    $entity_prophecy = $this->prophesize(ContentEntityInterface::class);
+
+    $entity_prophecy->getFieldDefinition($field_name)
+      ->willReturn($field_definition_prophecy->reveal());
+
+    // If the cardinality is unlimited getting a cound of the field items is
+    // never expected, so just check it's not called.
+    if ($cardinality === FieldStorageDefinitionInterface::CARDINALITY_UNLIMITED) {
+      $entity_prophecy->get($field_name)
+        ->shouldNotBeCalled();
+    }
+    // Otherwise, there will be a count retrieved from the field item.
+    else {
+      $field_item_prophecy = $this->prophesize(FieldItemListInterface::class);
+      $field_item_prophecy->count()
+        ->willReturn($field_count)
+        ->shouldBeCalled();
+
+      $entity_prophecy->get($field_name)
+        ->willReturn($field_item_prophecy->reveal())
+        ->shouldBeCalled();
+    }
+
+    $this->assertSame(OgGroupAudienceHelper::checkFieldCardinality($entity_prophecy->reveal(), $field_name), $expected);
+  }
+
+  /**
+   * Data provider for testFieldCardinality.
+   *
+   * @return array
```

Comment:
I'd list down the values its returning, as it makes it easier to read, rather then go back to `testFieldCardinality`.  maybe something like:

```php
@return array
  The values to test which correspond to:
  - The count of existing populated items in the field.
  - Field cardinality
  - The expected result where TRUE signifies the field may be populated by another value. 
```